lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -528,18 +530,31 @@ type installKubeAgentParams struct {
 	req          EnrollEKSClustersRequest
 }
 
-func getChartURL(version string) *url.URL {
+func getChartURL(version string) (*url.URL, error) {
+	repo := agentRepoURL
+	ver, err := semver.NewVersion(version)
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to parse chart version %q", version)
+	}
+
+	// pre release tagged charts are located in the staging repo.
+	if ver.PreRelease != "" {
+		repo = agentStagingRepoURL
+	}
 	return &url.URL{
-		Scheme: agentRepoURL.Scheme,
-		Host:   agentRepoURL.Host,
+		Scheme: repo.Scheme,
```

Comment:
Replace this copy with return `repo.JoinPath(fmt.Sprintf("%s-%s.tgz", agentName, version)), nil`